### PR TITLE
Fix windows callback verification #22

### LINKED ISSUE #22: Windows callbackUrlScheme are invalid
## Describe the bug
When you run your example app, or any other implementation, it is not possible to give a valid callbackUrlScheme for windows, as windows needs a scheme that matchs `RegEx()` and you check the scheme before with `RegEx`, it is not possible to start a windows authentication.

## To Reproduce
1. Go to 'example/lib/main.dart'
2. Run the app
3. Click on 'Authenticate'

## Expected behavior
It is possible to add a callbackUrlScheme for windows that matchs the RegEx from the plattform implementation.

## Device (please complete the following information!)
 - Device: Windows PC
 - OS: Windows
 - `flutter_web_auth_2` version: 2.0.1

## Additional context
/flutter_web_auth_2/lib/flutter_web_auth_2.dart in line 62 and flutter_web_auth_2/lib/src/flutter_web_auth_2_windows.dart line 66 close each other.

## Fix
#23 for the fix

### PR DESCRIPTION
**Pull request by Knupper**
use _schemeRegExp not for windows platform
update url_launcher
add changelog
Fixes #22

### CODE REVIEW

**Comment by Knupper** [comment]
WIP

**Comment by ThexXTURBOXx** [comment]
I have pushed a commit to your branch. Could you please verify if that still works for you?
Using funcitonality from `dart:io` throws Exceptions on web.
This will now check the platform itself after confirming it is not web, but the assertion is still checked on every platform (in your fix, the assertion was only checked on web, not anymore on the other platforms)

File: flutter_web_auth_2/lib/flutter_web_auth_2.dart [hunk 1]
```diff
@@ -31,6 +32,17 @@ class FlutterWebAuth2 {
   static final _OnAppLifecycleResumeObserver _resumedObserver =
       _OnAppLifecycleResumeObserver(_cleanUpDanglingCalls);
 
+  static void _assertCallbackScheme(String callbackUrlScheme) {
+    if (!_schemeRegExp.hasMatch(callbackUrlScheme) &&
+        (kIsWeb || !Platform.isWindows)) {
```
[the remark this answers is not in the document]
Comment (Knupper) [reply flutter_web_auth_2/lib/flutter_web_auth_2.dart:37]: Okay, that was my first assumption as well. i checked it again, and your code is working. Maybe it was to much yesterday.

[the remark this answers is not in the document]
Comment (ThexXTURBOXx) [reply flutter_web_auth_2/lib/flutter_web_auth_2.dart:37]: Perfect, no problem! :)

**Comment by ThexXTURBOXx** [comment]
`2.0.2` has just been released! Thanks again for this PR! :)

**Pull request merged**